File: src/main/java/org/oyster/card/rule/FareRule.java
package org.oyster.card.rule;

import org.oyster.card.dto.JourneyType;
import org.oyster.card.dto.Zone;

import java.math.BigDecimal;

public interface FareRule {
    boolean shouldProcess(JourneyType type, Zone one, Zone two);
    BigDecimal getFare();
}

File: kotlin-impl/src/main/kotlin/org/oyster/card/rule/FareRule.kt
package org.oyster.card.rule

import org.oyster.card.dto.JourneyType
import org.oyster.card.dto.Zone
import java.math.BigDecimal

interface FareRule {

    fun shouldProcess(type :JourneyType, one :Zone, two :Zone) : Boolean
    fun getFare() : BigDecimal
}